sycl/test-e2e/Config/select_device.cpp

```diff
@@ -167,285 +167,215 @@ static std::vector<DevDescT> getAllowListDesc(std::string allowList) {
+                << "}}" << std::endl;
+      return 0;
+    } else {
+      sycl::queue deviceQueue(gpu_selector_v);
+      device dev = deviceQueue.get_device();
+      std::cout << "Device: " << dev.get_info<info::device::name>()
+                << std::endl;
+      std::cout << "DriverVersion: "
+                << dev.get_info<info::device::driver_version>() << std::endl;
+      for (const DevDescT &desc : components) {
+        if ((std::regex_match(dev.get_info<info::device::name>(),
+                              std::regex(desc.devName))) &&
+            (std::regex_match(dev.get_info<info::device::driver_version>(),
+                              std::regex(desc.devDriverVer)))) {
+          return 0;
         }
       }
-      fs.close();
+      return 1;
     }
-  } else if (getenv("READ_PLATFORM_INFO")) {
-    std::ifstream fs;
-    fs.open("select_device_config.txt", std::fstream::in);
-    if (fs.is_open()) {
-      std::string allowlist;
-      std::getline(fs, allowlist);
-      if (!allowlist.empty()) {
-        setenv("SYCL_DEVICE_ALLOWLIST", allowlist.c_str(), 0);
-        std::vector<DevDescT> components(getAllowListDesc(allowlist));
-        std::cout << "SYCL_DEVICE_ALLOWLIST=" << allowlist << std::endl;
+  }
 
-        sycl::queue deviceQueue(gpu_selector_v);
-        device dev = deviceQueue.get_device();
-        const auto &plt = dev.get_platform();
-        for (const DevDescT &desc : components) {
-          if ((std::regex_match(plt.get_info<info::platform::name>(),
-                                std::regex(desc.platName))) &&
-              (std::regex_match(plt.get_info<info::platform::version>(),
-                                std::regex(desc.platVer)))) {
-            passed = true;
-          }
+  // Test the platform name and version number.
+  if (test == "PLATFORM_INFO"sv) {
+    if (write) {
+      std::optional<platform> plt = get_test_platform(aspect::gpu) ;
+      if (!plt)
+        return 3;
+
+      std::string name = plt->get_info<info::platform::name>();
+      addEscapeSymbolToSpecialCharacters(name);
+      std::string ver = plt->get_info<info::platform::version>();
+      std::cout << "PlatformName:{{" << name << "}},PlatformVersion:{{" << ver
+                << "}}" << std::endl;
+      return 0;
+    } else {
+      sycl::queue deviceQueue(gpu_selector_v);
+      device dev = deviceQueue.get_device();
+      const auto &plt = dev.get_platform();
+      std::cout << "Platform: " << plt.get_info<info::platform::name>()
+                << std::endl;
+      std::cout << "Platform Version: "
+                << plt.get_info<info::platform::version>() << std::endl;
+      for (const DevDescT &desc : components) {
+        std::cout << "Desc pltName: " << desc.platName << std::endl;
+        std::cout << "Desc pltVer: " << desc.platVer << std::endl;
+        if ((std::regex_match(plt.get_info<info::platform::name>(),
+                              std::regex(desc.platName))) &&
```

Comment:
if the platform name contains any regex reserved character, like `.  ?  * [ ] ^ $ % \ -`  this won't work.   I haven't use the std:: regex library much, but in other languages there is "RegexEscape" or equivalent that lets you convert any string into an arbitrary "safe" regex that just matches the intended string.

Reply:
I'm not modifying this either.

Reply:
do we even need the regex then? Wouldn't just a standard string search  or position work?